tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_beacon.c

```diff
@@ -295,8 +295,10 @@ static bool wait_for_beacon(bool (*process_cb)(const uint8_t *net_id, void *ctx)
 		FAIL("starting scan failed (err %d)", err);
 	}
 
-	/* Listen to beacons ONLY for one beacon interval. */
-	err = k_sem_take(&observer_sem, BEACON_INTERVAL);
+	/* Listen to beacons ONLY for one beacon interval.
+	 * (BEACON_INTERVAL + 1) to avoid time races with advertiser.
```

Comment:
What time races are happening here? I think the sleep should be exactly 10 seconds. +1 second is needed for the legacy adv, but this is not the case for the extended adv. Can this be an implementation bug?

Reply:
If add timestamping for beacon rx time you will see that beacon is received in the boundary of 10 seconds expiration. Usually uptime delta is 9900 us between 10 seconds and time when test starts waiting beacon and when beacon is received the previous beacon. Test starts waiting beacon on the time when the previous beacon is received.
I'm a bit surprised that works for legacy advertiser.

Reply:
And yep, test fails not because beacon is not received. It fails because beacon is received when test doesn't expect beacon.

Reply:
Could you clarify that in the comment? Current comment gives inaccurate explanation.